### DISCUSSION TITLE
any method to download all files belonging owned by a user?

### DISCUSSION THREAD

**Discussion by devhandler** [supabase/supabase#15549] [Questions] [answered]
the storage.objects table has the owner column that shows which files belong to which owner.

but from the storage download api, it only allows you to download one file a time. moreover, you must know the file path beforehand. is there any way to download all files belong to a user/owner and use the owner column to filter instead of a file path?

![image](https://github.com/supabase/supabase/assets/133625693/ca04f80a-13b8-4fba-a0d3-cb361060d958)

**Comment by GaryAustin1** [comment, marked as answer]
No storage API call that I know of.  You can write an rpc function to return all paths belonging to a user then convert with the publicUrl helper if a public bucket, or call signedUrls to get those.
